core/src/main/scala/filodb.core/memstore/ratelimit/CardinalityTracker.scala

```diff
@@ -169,50 +167,22 @@ class CardinalityTracker(ref: DatasetRef,
   }
 
   /**
-   * Use this method to query the top-k cardinality consumers
-   *   under a provided shard key prefix.
+   * Use this method to query cardinalities under a provided shard key prefix.
    *
    * @param depth cardinalities are returned for all prefixes of this size
    * @param shardKeyPrefix zero or more elements that form a valid shard key prefix
-   * @return Top-K records, can the less than K if fewer children
    */
-  def topk(k: Int, shardKeyPrefix: Seq[String], depth: Int, addInactive: Boolean): Seq[CardinalityRecord] = {
+  def scan(shardKeyPrefix: Seq[String], depth: Int): Seq[CardinalityRecord] = {
     require(shardKeyPrefix.length <= shardKeyLen, s"Too many shard keys in $shardKeyPrefix - max $shardKeyLen")
     require(depth >= shardKeyPrefix.size,
       s"depth $depth must be at least the size of the prefix ${shardKeyPrefix.size}")
 
-    implicit val ord = new Ordering[CardinalityRecord]() {
-      override def compare(x: CardinalityRecord, y: CardinalityRecord): Int = {
-        if (addInactive) x.tsCount - y.tsCount
-        else x.activeTsCount - y.activeTsCount
-      }
-    }.reverse
-
     if (shardKeyPrefix.size == depth) {
-      // directly fetch the cardinality and return
-      val card = getCardinality(shardKeyPrefix)
-      return Seq(CardinalityRecord(
-        shard, card.prefix, card.tsCount, card.activeTsCount,
-        if (shardKeyPrefix.length == shardKeyLen - 1) card.tsCount else card.childrenCount,
```

Comment:
I removed this `if` and the one formerly at line 208. No tests fail, but I'm not 100% certain of why they were there. Should I push this `if` down into the `CardinalityStore`? Put them back here?